src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerExtensions.cs

```diff
@@ -120,7 +121,7 @@ internal static ILoggingBuilder AddConsoleWithFormatter<TOptions>(this ILoggingB
         /// Adds a custom console logger formatter 'TFormatter' to be configured with options 'TOptions'.
         /// </summary>
         /// <param name="builder">The <see cref="ILoggingBuilder"/> to use.</param>
-        public static ILoggingBuilder AddConsoleFormatter<TFormatter, TOptions>(this ILoggingBuilder builder)
+        public static ILoggingBuilder AddConsoleFormatter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TFormatter, TOptions>(this ILoggingBuilder builder)
```

Comment:
Nothing for TOptions?

Reply:
Not that I can see. This code only configures the `TOptions`, it doesn't create/require it. Once someone tries to get an `IOptions<TOptions>` (whatever the `TOptions` type is), they will be the ones preserving the constructor on it.

The typical place that someone is going to require the `TOptions` will be in the `TFormatter` class's constructor.